experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/client/Client.scala

```diff
@@ -189,8 +188,7 @@ class Client(configuration:AppConfig, yarnConf: YarnConfiguration, yarnClient: Y
     val appContext = yarnClient.createApplication.getApplicationSubmissionContext
     appContext.setApplicationName(getEnv(YARNAPPMASTER_NAME))
 
-    val containerContext: ContainerLaunchContext = YarnContainerUtil.getContainerContext(yarnConf, getCommand)
-    containerContext.setLocalResources(YarnContainerUtil.getAMLocalResourcesMap(yarnConf, getConfiguration))
+    val containerContext = DefaultContainerLaunchContextFactory(yarnConf, getConfiguration).newInstance(getCommand)
     appContext.setAMContainerSpec(containerContext)
     appContext.setResource(getAMCapability)
     appContext.setQueue(getEnv(YARNAPPMASTER_QUEUE))
```

Comment:
```
  def deploy() = {
    LOG.info("Starting AM")
    //uploadAMResourcesToHDFS()
    uploadConfigToHDFS()
    yarnClient.init(yarnConf)
    yarnClient.start()
    val appContext = yarnClient.createApplication.getApplicationSubmissionContext
    appContext.setApplicationName(getEnv(YARNAPPMASTER_NAME))

    val containerContext = DefaultContainerLaunchContextFactory(yarnConf, getConfiguration).newInstance(getCommand)
    appContext.setAMContainerSpec(containerContext)
    appContext.setResource(getAMCapability)
    appContext.setQueue(getEnv(YARNAPPMASTER_QUEUE))
    
    yarnClient.submitApplication(appContext)
    monitorAM(appContext)
  }

  deploy()
```

It maybe better to call deploy() when using Client, insteading of constructing Client.